### PR TITLE
feat: Issue state automation set up, add issue stream by system

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Issue state automation set up, add issue stream by system

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Issue state automation set up, add issue stream by system           |
| 🇨🇳 中文    |   事项状态流的自动化           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/dop/services/issue/issue.go [hunk 1]
```diff
@@ -692,18 +700,147 @@ func (svc *Issue) UpdateIssue(req apistructs.IssueUpdateRequest) error {
...
+	stateOld apistructs.IssueStateBelong
+	stateNew apistructs.IssueStateBelong
+}
+
+func (svc *Issue) AfterIssueChildrenUpdate(u *issueChildrenUpdated) error {
+	if u == nil {
+		return fmt.Errorf("issue children update config is empty")
+	}
+	parents, err := svc.db.GetIssueParents(u.id, []string{apistructs.IssueRelationInclusion})
+	if err != nil || len(parents) != 1 {
```
Comment (Effet) [inline modules/dop/services/issue/issue.go:714]: it is quite confusion, I suggest to change to:
```
if err != nil {
  return err
}
if l := len(parents); l != 1 {
  if l > 1 {
    // log warning
  }
  return nil
}
```

File: modules/dop/services/issue/issue.go [hunk 2]
```diff
@@ -692,18 +700,147 @@ func (svc *Issue) UpdateIssue(req apistructs.IssueUpdateRequest) error {
...
 		return err
 	}
+	p := parents[0]
+
+	if p.Belong == string(apistructs.IssueStateBelongOpen) && u.stateOld != "" &&
+		(u.stateOld != apistructs.IssueStateBelongOpen || u.stateNew != apistructs.IssueStateBelongOpen) {
```
Comment (Effet) [inline modules/dop/services/issue/issue.go:720]: what if u.stateOld == u.stateNew?

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/services/issue/issue.go (hunk 1 above)
Comment (shuofan) [reply modules/dop/services/issue/issue.go]: done

File: modules/dop/services/issue/issue.go [hunk 3]
```diff
@@ -657,7 +657,15 @@ func (svc *Issue) UpdateIssue(req apistructs.IssueUpdateRequest) error {
 	}
 
 	if issueModel.Type == apistructs.IssueTypeTask {
-		if err := svc.AfterIssueChildrenUpdate(issueModel.ID); err != nil {
+		currentBelong, err := cache.TryGetState(issueModel.State)
+		if err != nil || currentBelong == nil {
+			return apierrors.ErrUpdateIssue.InternalError(err)
```
Comment (Effet) [inline modules/dop/services/issue/issue.go:662]: if err == nil, the error message will empty, I suggest to compose a new error

File: modules/dop/services/issue/issue.go [hunk 4]
```diff
@@ -657,7 +657,15 @@ func (svc *Issue) UpdateIssue(req apistructs.IssueUpdateRequest) error {
...
+		}
+		newBelong, err := cache.TryGetState(*req.State)
+		if err != nil || newBelong == nil {
+			return apierrors.ErrUpdateIssue.InternalError(err)
```
Comment (Effet) [inline modules/dop/services/issue/issue.go:666]: here too

**Review by Effet — CHANGES_REQUESTED** [review]
we will merge after you finish the last change request, and post the picture in dev ❤️

File: modules/dop/services/issue/issue.go (hunk 3 above)
Comment (shuofan) [reply modules/dop/services/issue/issue.go]: done

File: modules/dop/services/issue/issue.go (hunk 4 above)
Comment (shuofan) [reply modules/dop/services/issue/issue.go:669]: image

**Review by Effet — APPROVED** [review]

**Pull request merged**